File: model/src/main/java/net/contrapt/auction/model/Purchase.java
package net.contrapt.auction.model;

import javax.persistence.Entity;
import javax.persistence.Table;
import java.math.BigDecimal;

/**
 * Created by msimmons on 11/11/14.
 */
@Entity
@Table(name = "purchase")
public class Purchase extends AbstractEntity {

    private Long bidderId;

    private Long productId;

    private BigDecimal amount;

    @Override
    public String uniqueKey() {
        return bidderId+":"+productId+":"+getId();
    }
}

File: model/src/main/kotlin/net/contrapt/auction/model/Purchase.kt
package net.contrapt.auction.model

import java.math.BigDecimal
import javax.persistence.Entity
import javax.persistence.Table

/**
 * Created by msimmons on 11/11/14.
 */
@Entity
@Table(name = "purchase")
class Purchase(val bidderId : Long, val productId : Long, val amount : BigDecimal) : AbstractEntity() {


    override fun uniqueKey(): String {
        return "$bidderId:$productId:$id"
    }
}
